shelley/chain-and-ledger/formal-spec/multi-sig.tex

```diff
@@ -0,0 +1,567 @@
+  to spend the output.
+\end{abstract}
+
+% \section*{List of Contributors}
+% \label{acknowledgements}
+
+% \tableofcontents
+% \listoffigures
+
+\section{Introduction}
+\label{sec:introduction}
+
+This specification for a simple multi-signature scheme is based
+on~\cite{shelley_formal_spec} which formally specifies the Shelley Cardano
+ledger. The main changes are the following:
+
+\begin{itemize}
+\item Add a new address type that for outputs locked by scripts.
+\item Change the type of transaction inputs of a transaction to optionally
+  include a validator script.
+\item Adapt the transaction validation in such a way that funds locked by a
+  multi-signature script can be spent.
+\item Adapt the functions used in the validation for the extended types of
+  addresses and transaction inputs with scripts.
+\end{itemize}
+
+In this approach for multi-signature, the scripts receive a set of hashed keys
+where each corresponding key was used to correctly sign the transaction. The
+script can then simply check the hashes against its own representation of which
+keys in which combination can unlock the unspent output.
+
+This means that the scripts are completely stateless and no data needs to be
+supplied to the script apart from the hashed keys used to sign the transaction.
+It allows for any type of $n$ out of $m$ required signatures for a specific
+unspent transaction output.
+
+\section{Types}
+\label{sec:types}
+
+\begin{figure*}[hbt]
+  \emph{Abstract types}
+
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      script & \Script & \text{script type}
+    \end{array}
+  \end{equation*}
+
+  \emph{Derived types}
+
+  \begin{equation*}
+    \begin{array}{r@{~\in~}l@{\qquad=\qquad}lr}
+      \var{addr} & \AddrScr & \HashScr \times \Script & \text{Script address} \\
+      \var{addr} & \Addr & \begin{array}{l@{~\uniondistinct}l}
+                             \AddrB & \AddrP \uniondistinct \AddrE \\
+                                    & \AddrBS \uniondistinct \AddrScr
+                           \end{array}
+                            & \text{output address}
+    \end{array}
+  \end{equation*}
+
+  \emph{Accessor Functions}
+
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \fun{paymentHK} & \Addr \to \HashKey_{pay}^{?}
+      & \text{hash of payment key from addr}\\
+      \fun{validatorHash} & \Addr \to \HashScr^{?} & \text{hash of validator script}
+    \end{array}
+  \end{equation*}
+
+  \emph{Abstract Functions}
+
+  \begin{equation*}
+    \begin{array}{{r@{~\in~}lr}}
+      \fun{hashScript} & \Script \to \HashScr & \text{hash a script}
+    \end{array}
+  \end{equation*}
+
+  \caption{Types for Scripts and Script Addresses}
+  \label{fig:types-scripts}
+\end{figure*}
+
+In Figure~\ref{fig:types-scripts} the $\Addr$ type of~\cite{shelley_formal_spec}
+is extended to also include script addresses. This new script address comprises
+the hash of the validator script an the content of a data script.
```

Comment:
I think you didn't intend to write "an the content of a data script." here

Reply:
good catch, should be `and`

Reply:
But the addresses introduced here don't have a data script. Or by "content of a data script" do you mean that the validator script in question contains all the data that it needs already?

Reply:
that is even more true, I had thought you referred to the typo. That phrase was a remnant of the original text of that paragraph which was for extended UTxO.

Reply:
thanks for the review @JaredCorduan should be ok now